src/Common/DateLUTImpl.h

```diff
@@ -1030,7 +1030,7 @@ class DateLUTImpl
     template <typename DateOrTime>
     DateOrTime toStartOfMinuteInterval(DateOrTime t, UInt64 minutes) const
     {
-        UInt64 divisor = 60 * minutes;
+        Int64 divisor = 60 * minutes;
```

Comment:
Hm, why?

Reply:
Because unsigned is promoted in C and C++.
Consider 'Time res = date + (t - date) / divisor * divisor' , for example.

Reply:
We found the bug that functions like toStartOfFiveMinutes (which are using toStartOfMinuteInterval) did work well for 1970 and later dates. But they did not work at all for dates before 1970. This is because t is negative. And division like:
negative t by unsigned divisor produced a "garbage".

Consider the example:
```
int main()
{
    unsigned int divisor = 60;
    int t = -90;
    cout << t / divisor << endl;
    return 0;
}
```
It returns 71582786.

But division of signed int by signed int works fine:
```
int main()
{
    int divisor = 60;
    int t = -90;
    cout << t / divisor << endl;
    return 0;
}
```
It returns -1.

Reply:
I see, thanks. No objections then. It would be nice though if we had tests for the edge cases, i.e. input values close to the min/max date/time range.